app/services/order_syncer.rb

```diff
@@ -66,12 +66,16 @@ def update_payment_for(order)
   end
 
   def update_shipment_for(order)
-    shipment = order.shipments.with_state('pending').where(shipping_method_id: shipping_method_id_was).last
+    shipment = pending_shipment_using_shipping_method(order, shipping_method_id_was)
+
     if shipment
```

Comment:
maybe this can simply be
`if order.shipment.shipping_method == shipping_method_id_was?`